tests/integration-tests/tests/networking/test_security_groups.py

```diff
@@ -45,21 +45,37 @@ def test_additional_sg_and_ssh_from(region, custom_security_group, pcluster_conf
     check_head_node_security_group(region, cluster, 22, ssh_from)
 
 
+@pytest.mark.parametrize("additional_security_groups", [False, True])
 @pytest.mark.usefixtures("os", "instance")
-def test_overwrite_sg(region, scheduler, custom_security_group, pcluster_config_reader, clusters_factory):
+def test_overwrite_sg(
+    region, scheduler, custom_security_group, pcluster_config_reader, clusters_factory, additional_security_groups
+):
     """Test vpc_security_group_id overwrites pcluster default sg on head and compute nodes, efs, fsx"""
-    custom_security_group_id = custom_security_group.cfn_resources["SecurityGroupResource"]
+    count_of_sgs = 2
+
+    custom_security_group_stack = custom_security_group(count_of_sgs=count_of_sgs)
```

Comment:
I followed the same modularity which was being followed earlier w.r.t `custom_security_group`. But then after changing the nos of outputs that it gives, it makes sense to return a list rather than a stack. Will make the change as you suggest